.github/workflows/ci.yml

```diff
@@ -0,0 +1,100 @@
+name: CI
+
+on:
+  pull_request:
+
+env:
+  JAVA_TOOL_OPTIONS: -Xmx4096m
+  GRADLE_OPTS: -Dorg.gradle.daemon=false -Dorg.gradle.parallel=true -Dorg.gradle.workers.max=4 -Xmx4096m
+
+jobs:
+  build:
+    # 4 cpu, 16G ram
+    runs-on: ubuntu-24.04
+    environment: dev
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Prepare
+        uses: ./.github/actions/prepare
+
+      - name: Build and unit test
+        run: |
+          ./gradlew build
+
+      - name: Integration Test
+        run: |
+          ./gradlew integrationTest
+
+      - name: Test results and report
+        uses: ./.github/actions/testResultsReports
+        with:
+          suiteName: 'unitIntegrationTests'
+
+      - name: Upload workspace build
+        uses: actions/upload-artifact@v4
+        with:
+          name: workspace
+          path: |
+            ./
+            !./.git/**
+          retention-days: 7
+
+  acceptanceTests:
+    needs: build
+    # 4 cpu, 16G ram
+    runs-on: ubuntu-24.04
+    environment: dev
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Prepare
+        uses: ./.github/actions/prepare
+
+      - name: Download workspace build
+        uses: actions/download-artifact@v4
+        with:
+          name: workspace
+
+      - name: Acceptance Tests
+        run: |
+          ls -la .
+          if [ -n "$GCP_SERVICE_KEY" ]; then
```

Comment:
we should not need these in CI PR Build.

Reply:
Also, this is a bit of security issue because currently we are storing the whole workspace as an artifact.